pages/docs/reference/compiler-reference.md

```diff
@@ -0,0 +1,356 @@
+
+### `-module-kind {plain|amd|commonjs|umd}`
+
+- Kind of the JS module generated by the compiler:
+    - `plain` - a plain JS module;
+    - `commonjs` - a [CommonJS](http://www.commonjs.org/) module;
+    - `amd` - an [Asynchronous Module Definition](https://en.wikipedia.org/wiki/Asynchronous_module_definition) module;
+    - `umd` - a [Universal Module Definition](https://github.com/umdjs/umd) module.
+    
+    To learn more about the module kinds and dictinctions between them,
+    see [this](https://www.davidbcalhoun.com/2014/what-is-amd-commonjs-and-umd/) article.
+    
+### `-no-stdlib`
+
+- Exclude `kotlin-stdlib-js.jar` from the compilation dependencies.
+
+### `-output <filepath>`
+
+- Set the destination file for the compilation result. The value must be a path to a `.js` file including its name.
+
+### `-output-postfix <filepath>`
+
+- Add the content of the specified file to the end of output file.
+
+### `-output-prefix <filepath>`
+
+- Add the content of the specified file to the beginning of output file.
+
+### `-source-map`
+
+- Generate the source map.
+
+### `-source-map-base-dirs <path>`
+
+- Use the specified paths as base directories. Base directories are used for calculating relative paths in the source map.
+
+### `-source-map-embed-sources {always|never|inlining}`
+
+- Embed source files into the source map.
+
+### `-source-map-prefix`
+
+- Add the specified prefix to paths in the source map.
+
+
+## Kotlin/Native compiler options
+
+Kotlin/Native compiler compiles Kotlin source files into native binaries for [supported platforms](native-overview.html#target-platforms). 
+The command-line tool for Kotlin/Native compilation is `kotlinc-native`.
+
+In addition to [common options](#common-options), Kotlin/Native compiler has the options listed below.
+
+
+### `-enable-assertions`
+
+- Enable runtime assertions in the generated code.
+    
+### `-g`
+
+- Enable emitting debug information.
+    
+### `-generate-test-runner` (`-tr`)
+
+- Produce a runner for unit tests.
+    
+### `-generate-worker-test-runner` (`-trw`)
+
+- Produce a worker runner for unit tests.
+    
+### `-generate-no-exit-test-runner` (`-trn`)
+
+- Produce a runner for unit tests not forcing exit.
+    
+### `-include-binary <path>` (`-ib <path>`)
+
+- Pack external binary within the generated klib file.
+    
+### `-library <path>` (`-l <path>`)
+
+- Link with the library.
+
+### `-library-version <version>` (`-lv`)
+
+- Set library version.
+    
+### `-list-targets`
+
+- List available hardware targets.
+
+### `-manifest <path>`
+
+- Provide a manifest addend file.
+    
+### `-memory-model {strict|relaxed}`
+
+- Memory model to use. Two memory models are supported:
+    - _strict_
+    - _relaxed_
+
+### `-module-name <name>`
+
+- Specify a name for the compilation module.
+
+### `-native-library <path>`(`-nl <path>`)
+
+- Include the native bitcode library.
+
+### `-no-default-libs`
+
+- Don't link the libraries from dist/klib automatically.
```

Comment:
Description of this flag is obsolete and should be update at the compiler side. Currently it disables linking a user code with default platform libraries distributed with the compiler (More about platform libraries: https://kotlinlang.org/docs/reference/native/platform_libs.html).